From 5ee25d0ac7a82f3fb7a8a98603143939f0a1c19e Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:03:03 -0400
Subject: [usb] wusbcore: set the RPIPE wMaxPacketSize value correctly

Message-id: <1410887463-20674-112-git-send-email-dzickus@redhat.com>
Patchwork-id: 95489
O-Subject: [RHEL7 PATCH 111/591] usb: wusbcore: set the RPIPE wMaxPacketSize value correctly
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 7b6bc07ab554e929c85d51b3d5b26cf7f12c6a3b
Author: Thomas Pugliese <thomas.pugliese@gmail.com>
Date:   Wed Oct 23 14:44:26 2013 -0500

    usb: wusbcore: set the RPIPE wMaxPacketSize value correctly

    For isochronous endpoints, set the RPIPE wMaxPacketSize value using
    wOverTheAirPacketSize from the endpoint companion descriptor instead of
    wMaxPacketSize from the normal endpoint descriptor.

    Signed-off-by: Thomas Pugliese <thomas.pugliese@gmail.com>
    Cc: stable <stable@vger.kernel.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/wusbcore/wa-rpipe.c b/drivers/usb/wusbcore/wa-rpipe.c
index 1ed068a..b48e74c 100644
--- a/drivers/usb/wusbcore/wa-rpipe.c
+++ b/drivers/usb/wusbcore/wa-rpipe.c
@@ -334,7 +334,10 @@ static int rpipe_aim(struct wa_rpipe *rpipe, struct wahc *wa,
  /* FIXME: compute so seg_size > ep->maxpktsize */
  rpipe->descr.wBlocks = cpu_to_le16(16);  /* given */
  /* ep0 maxpktsize is 0x200 (WUSB1.0[4.8.1]) */
- rpipe->descr.wMaxPacketSize = cpu_to_le16(ep->desc.wMaxPacketSize);
+ if (usb_endpoint_xfer_isoc(&ep->desc))
+  rpipe->descr.wMaxPacketSize = epcd->wOverTheAirPacketSize;
+ else
+  rpipe->descr.wMaxPacketSize = ep->desc.wMaxPacketSize;
 
  rpipe->descr.hwa_bMaxBurst = max(min_t(unsigned int,
     epcd->bMaxBurst, 16U), 1U);
-- 
1.7.1